src/main/java/de/tum/in/www1/artemis/web/rest/ModelingAssessmentResource.java

```diff
@@ -150,11 +160,22 @@ public ModelingAssessmentResource(JsonAssessmentRepository jsonAssessmentReposit
         return ResponseEntity.ok("");
     }
 
+    /**
+     * Saves assessments and updates result.
+     *
+     * @param exerciseId            the exerciseId the assessment belongs to
+     * @param resultId              the resultId the assessment belongs to
+     * @param modelingAssessment    the assessments as string
+     * @return the ResponseEntity with result as body
+     */
     @PutMapping("/modeling-assessments/exercise/{exerciseId}/result/{resultId}")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Timed
     public ResponseEntity<Result> saveModelingAssessment(@PathVariable Long exerciseId, @PathVariable Long resultId, @RequestBody String modelingAssessment) {
         ModelingExercise modelingExercise = modelingExerciseService.findOne(exerciseId);
```

Comment:
security missing: I think we should check if the TA or instructor are in the course